core/src/main/java/org/mobilitydata/gtfsvalidator/validator/ValidationContext.java

```diff
@@ -16,26 +16,33 @@
 
 package org.mobilitydata.gtfsvalidator.validator;
 
-import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
 import org.mobilitydata.gtfsvalidator.input.CountryCode;
 import org.mobilitydata.gtfsvalidator.input.CurrentDateTime;
 
 /**
  * A read-only context passed to particular validator objects. It gives information relevant for
  * validation: properties of the feed as a whole, system properties (current time) etc.
  */
-@AutoValue
-public abstract class ValidationContext {
+public class ValidationContext {
   public static Builder builder() {
-    return new AutoValue_ValidationContext.Builder();
+    return new Builder();
+  }
+
+  private final ImmutableMap<Class<?>, Object> context;
+
+  private ValidationContext(ImmutableMap<Class<?>, Object> context) {
+    this.context = context;
   }
 
   /**
    * Represents a name of a GTFS feed, such as "nl-openov".
    *
    * @return the @code{GtfsFeedName} representing the feed's name
```

Comment:
Unrelated to your PR, but this method comment seems incorrect.  Understood if you don't want to fix it here, but if you were willing.

Reply:
Let's fix it while I am here. This comment is pretty confusing.